c7n/resources/elasticsearch.py

```diff
@@ -588,3 +588,41 @@ class resource_type(TypeInfo):
         filter_type = 'list'
         arn_type = "reserved-instances"
         permissions_enum = ('es:DescribeReservedElasticsearchInstances',)
+
+
+@ElasticSearchDomain.action_registry.register('update-tls-config')
+class UpdateTlsConfig(Action):
+
+    """Action to update tls-config on a domain endpoint
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: update-tls-config
+                resource: elasticsearch
+                filters:
+                  - type: value
+                    key: 'DomainEndpointOptions.TLSSecurityPolicy'
+                    op: eq
+                    value: "Policy-Min-TLS-1-0-2019-07"
+                actions:
+                  - type: update-tls-config
+                    value: "Policy-Min-TLS-1-2-2019-07"
+    """
+
+    schema = type_schema('update-tls-config', value={'type': 'string',
+        'enum': ['Policy-Min-TLS-1-0-2019-07', 'Policy-Min-TLS-1-2-2019-07']}, required=['value'])
+    permissions = ('es:UpdateElasticsearchDomainConfig', 'es:DescribeElasticsearchDomainConfig',
```

Comment:
only the permission for the action is needed, the resource manager handles the permissions it needs to fetch resources.

Reply:
fixed.